src/routes/DisclaimerPage/index.js

```diff
@@ -2,38 +2,42 @@ import React from 'react'
 import PropTypes from 'prop-types'
 
 import { connect } from 'react-redux'
+
 import Page from 'components/Content/Page'
 import RichLayout from 'components/RichLayout'
-import { DisclaimerFetcher } from 'endpoints'
+import PageModel from 'endpoints/models/PageModel'
+import withFetcher from 'endpoints/withFetcher'
+import DISCLAIMER_ENDPOINT from 'endpoints/disclaimer'
+
+class ContentWrapper extends React.Component {
+  static propTypes = {
+    disclaimer: PropTypes.instanceOf(PageModel) // From withFetcher
+  }
 
-class PageAdapter extends React.Component {
   render () {
     return <Page page={this.props.disclaimer}/>
   }
 }
 
+const FetchingContentWrapper = withFetcher(DISCLAIMER_ENDPOINT)(ContentWrapper)
+
 class DisclaimerPage extends React.Component {
   static propTypes = {
-    language: PropTypes.string.isRequired,
-    location: PropTypes.string.isRequired
+    location: PropTypes.string.isRequired,
+    disclaimer: PropTypes.instanceOf(PageModel)
   }
 
   render () {
     return (
       <RichLayout location={this.props.location}>
-        <DisclaimerFetcher>
-          <PageAdapter/>
-        </DisclaimerFetcher>
+        <FetchingContentWrapper/>
       </RichLayout>
     )
   }
 }
 
 function mapStateToProps (state) {
-  return {
-    language: state.router.params.language,
-    location: state.router.params.location
-  }
+  return {location: state.router.params.location}
```

Comment:
We don't need location in the DisclaimerPage props, RichLayout is connected to the state itself.
So we disconnect the DisclaimerPage from the state 🎉

Reply:
done